_posts/2015-07-20-third-party-integration.md

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "Challenges of integrating third-parties"
+date: "2015-07-20 07:20:17"
+icon: cloud
+tags: integration third-party API webservice
+categories: tech-stories
+authors: ["Ulf B."]
+---
+
+Third-party applications are an important part of the service that ePages offers.
+As our applications are built on an internal application platform, ePages developers are highly dependent on third-party services.
+Read further, why it is not the easiest thing to integrate third-parties and how we tackle this.
+
+# The use case
+
+{% image blog-third-party-overview.png %}{% endimage %}
+
+Example: A customer buys something. They could choose e.g. PayPal as payment method and DHL as delivery method.
+The customer could also obtain an insurance with Trusted Shops.
+(Read further [here](http://www.epages.com/en/partner/technology-partner/) to get an overview of our technology partners.)
+A checkout alone can easily have three third-party dependencies.
+
+The merchant, in turn, receives information via server-sided callbacks, if an order has been already paid.
+They can generate shipping labels via the web service of a third-party shipping method or synchronise his online shop with ERP-software.
+These are just a few examples of third-party integrations at ePages.
+There are a lot more uses cases where both customer and merchant can use third-party services.
+
+Payment, deliveries, market places, marketing or tax calculation - for many applications, specialised services are available that have to be integrated into the platform at various locations.
+In every of these areas new startup companies are founded that offer new services.
+At the same time, also existing companies expand their service portfolio in order not to lose the connection in the dynamic e-commerce market.
+These circumstances keep us busy not only in integrating periodically new services but also in maintaining the existing integrations.
+
+If companies expand abroad, as it is the case at ePages, it might be required to include additional services, to being attractive for the merchants in that specific country.
+We integrated USPS and FedEx, since it would not make any sense to enter the US market without an integration of the most important US shipping agents.
```

Comment:
I would use another word for "most important" that it is not seen as we would rate our partners. Maybe something like "common", "widespread" which make clear that the customers are used and kind of expect to use this shipping agent.